platform_secrets/service.py

```diff
@@ -74,30 +79,57 @@ async def add_secret(self, secret: Secret) -> None:
             raise ValueError(f"Secret key {secret.key!r} or its value not valid")
 
     async def remove_secret(self, secret: Secret) -> None:
-        secret_name = self._get_kube_secret_name(secret.owner, secret.org_name)
+        secret_name = self._get_kube_secret_name(secret.project_name, secret.org_name)
         try:
             await self._kube_client.remove_secret_key(secret_name, secret.key)
         except (ResourceNotFound, ResourceInvalid):
             raise SecretNotFound.create(secret.key)
 
-    async def get_all_secrets(self, with_values: bool = False) -> list[Secret]:
-        payload = await self._kube_client.list_secrets()
+    async def get_all_secrets(
+        self,
+        with_values: bool = False,
+        org_name: Optional[str] = None,
+        project_name: Optional[str] = None,
+    ) -> list[Secret]:
+        label_selectors = []
+        if org_name:
+            label_selectors += [f"{SECRET_API_ORG_LABEL}={org_name}"]
+        if project_name:
+            label_selectors += [f"{PROJECT_LABEL}={project_name}"]
+        label_selector = ",".join(label_selectors) if label_selectors else None
+        payload = await self._kube_client.list_secrets(label_selector)
         result = []
         for item in payload:
-            owner, org_name = self._get_org_owner_from_secret_name(
-                item["metadata"]["name"],
-                org_name=item["metadata"].get("labels", {}).get(SECRET_API_ORG_LABEL),
-            )
-            if not owner:
+            labels = item["metadata"].get("labels", {})
+            owner = None
+            org_name = labels.get(SECRET_API_ORG_LABEL)
+            project_name = labels.get(PROJECT_LABEL)
+            if not project_name:
+                owner = self._get_owner_from_secret_name(
+                    item["metadata"]["name"], org_name
+                )
+                project_name = owner
+            if not project_name:
                 continue
             if with_values:
                 result += [
-                    Secret(key=key, value=value, owner=owner, org_name=org_name)
+                    Secret(
+                        key=key,
+                        value=value,
```

Comment:
minor: we probably could do `value=value if with_values else ""` end remove if-else